Modules/_ssl.c

```diff
@@ -2440,6 +2467,195 @@ PySSL_select(PySocketSockObject *s, int writing, PyTime_t timeout)
     return rc == 0 ? SOCKET_HAS_TIMED_OUT : SOCKET_OPERATION_OK;
 }
 
+/*[clinic input]
+_ssl._SSLSocket.uses_ktls_for_send
+
+Check if the Kernel TLS data-path is used for sending.
+[clinic start generated code]*/
+
+static PyObject *
+_ssl__SSLSocket_uses_ktls_for_send_impl(PySSLSocket *self)
+/*[clinic end generated code: output=f9d95fbefceb5068 input=604d98b67c65e8a7]*/
+{
+#ifdef BIO_get_ktls_send
+    int uses = BIO_get_ktls_send(SSL_get_wbio(self->ssl));
+    // BIO_get_ktls_send() returns 1 if kTLS is used and 0 if not.
+    // Also, it returns -1 for failure before OpenSSL 3.0.4.
+    if (uses == 1) {
+        Py_RETURN_TRUE;
+    }
+    else {
+        Py_RETURN_FALSE;
+    }
+#else
+    Py_RETURN_FALSE;
+#endif
+}
+
+/*[clinic input]
+_ssl._SSLSocket.uses_ktls_for_read
```

Comment:
I didn't spot this but I think it's perferrable to call it `for_recv` instead of `for_read`